File: leoz-node/src/main/java/org/deku/leoz/node/rest/WebserviceApplication.java
package org.deku.leoz.node.rest;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import javax.ws.rs.core.Application;
import java.util.HashSet;
import java.util.Set;

/**
 * Created by masc on 12.05.15.
 */
public class WebserviceApplication extends Application {
    private Logger mLog = LoggerFactory.getLogger(WebserviceApplication.class);
    private Set<Object> mSingletons = new HashSet<Object>();
    private Set<Class<?>> mClasses = new HashSet<>();

    public WebserviceApplication() {
        mLog.info("Leoz node jax/ws/rs webservice application");
    }

    @Override
    public Set<Object> getSingletons() {
        return mSingletons;
    }

    @Override
    public Set<Class<?>> getClasses() {
        return mClasses;
    }
}

File: leoz-node/src/main/java/org/deku/leoz/node/rest/WebserviceApplication.kt
package org.deku.leoz.node.rest

import org.jboss.resteasy.plugins.interceptors.CorsFilter
import org.slf4j.LoggerFactory

import javax.ws.rs.core.Application
import java.util.HashSet

/**
 * Created by masc on 12.05.15.
 */
class WebserviceApplication : Application() {
    private val log = LoggerFactory.getLogger(WebserviceApplication::class.java)
    private val singletons = HashSet<Any>()
    private val classes = HashSet<Class<*>>()

    init {
        log.info("Leoz node jax/ws/rs webservice application")
        val corsFilter = CorsFilter()
        corsFilter.allowedOrigins.add("*")
        corsFilter.allowedMethods = "OPTIONS, GET, POST, DELETE, PUT, PATCH"
        singletons.add(corsFilter)
    }

    override fun getSingletons(): Set<Any> {
        return singletons
    }

    override fun getClasses(): Set<Class<*>> {
        return classes
    }
}
